includes/ECommerce.php

```diff
@@ -164,4 +165,37 @@ public function register_assets() {
 			);
 		}
 	}
+
+	/**
+	 * Ensure that any retired BN codes are not sent with outbound requests to Paypal
+	 *
+	 * @param array  $parsed_args An array of HTTP request arguments
+	 * @param string $url         The request URL
+	 *
+	 * @return array Array of modified HTTP request arguments
+	 */
+	public function replace_retired_bn_codes( $parsed_args, $url ) {
+		try {
+			// Bail early if the request is not to paypal's v2 checkout API
+			if ( false === stripos( wp_parse_url( $url, PHP_URL_HOST ), 'paypal.com' )
+				&& false === stripos( wp_parse_url( $url, PHP_URL_PATH ), 'v2/checkout' ) ) {
+				return $parsed_args;
+			}
+
+			// Check for an existing bn_code and normalize to uppercase
+			$bn_code = isset( $parsed_args['headers']['PayPal-Partner-Attribution-Id'] ) ? strtoupper( $parsed_args['headers']['PayPal-Partner-Attribution-Id'] ) : null;
+
+			// Ensure we only set when blank, or when using one of our stale codes (not the current one)
+			if ( is_null( $bn_code ) ||
```

Comment:
Do we not want to set when blank? It looks like Mike's version did add our code when `!isset`

Reply:
This does set it if blank. 

Basically set the code if:
- is_null
- is not the new bn code (to prevent unnecessary replacement) **_AND_** has either `YITH` or `NEWFOLD` in it

Reply:
I suppose we could do an `empty()` instead of `isset()` check a few lines up in order to catch if it somehow had an empty string or a value of 0

Reply:
Ha, I see it now, I must have read it backward or missed that `||`.